pkg/apis/pravega/v1alpha1/pravegacluster_types.go

```diff
@@ -67,10 +67,9 @@ type ClusterSpec struct {
 	// available at: https://github.com/pravega/zookeeper-operator
 	ZookeeperUri string `json:"zookeeperUri"`
 
-	// ExternalAccess specifies whether or not to allow external access
-	// to clients and the service type to use to achieve it
+	// ExternalAccessEnabled boolean value specifies whether or not external access is enabled
 	// By default, external access is not enabled
-	ExternalAccess *ExternalAccess `json:"externalAccess"`
+	ExternalAccessEnabled bool `json:"externalAccessEnabled,omitempty"`
```

Comment:
This field seems unnecessary/redundant.   It is also highly unusual for CRDs.

Reply:
This field is necessary as it allows users to enable/disable external access at "Pravega Cluster" level instead of enabling it for individual components (partial external access)